lib/controllers/ephemeral/pin_entry_controller.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:async';
+import 'package:flutter/cupertino.dart';
+import 'package:get/get.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:pin_code_fields/pin_code_fields.dart';
+import 'package:pispapp/controllers/ephemeral/local_auth_controller.dart';
+import 'package:pispapp/routes/app_routes.dart';
+import 'package:pispapp/utils/log_printer.dart';
+
+class PINEntryController extends GetxController {
+  // Constructor to fetch PIN automatically
+  PINEntryController() {
+    _fetchPIN();
+  }
+
+  // Constants
+  static const String key = 'USER_PIN';
+  static const Duration timeoutDuration = Duration(milliseconds: 1000);
+  static const int PINlength = 6;
+  static final logger = getLogger('PinEntryController');
+
+  // Storage
+  final FlutterSecureStorage _storage = const FlutterSecureStorage();
```

Comment:
do you think it would be better to have a wrapper for this so that we can replace it easily without depending on this specific library?

Reply:
agreed, I will do the refactor.
It will be useful for the crypto stuff I'm working on as well